xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml

```diff
@@ -176,12 +176,24 @@ require(['jquery', 'xwiki-l10n!xwiki-confluence-attachments-messages'], function
     enhanceUploadInputs();
   })
 
+  $(document).on('xwiki:dom:updated', function(e, data) {
+    enhanceUploadInputs();
```

Comment:
Shouldn't you enhance only the part of the DOM that was updated? Or at least add a comment explaining why we enhance all inputs, including those that haven't been updated (and which may already be enhanced).